krummstab/summaries.py

```diff
@@ -0,0 +1,395 @@
+    alternately to gray for better readability.
+    """
+    first_name, last_name = name
+    if row % 2 == 0:
+        worksheet.write(row, 0, first_name, workbook.add_format(BORDER_TOP_BOTTOM | GRAY))
+        worksheet.write(row, 1, last_name, workbook.add_format(BORDER_TOP_BOTTOM | GRAY))
+        worksheet.set_row(row, None, workbook.add_format(GRAY))
+    else:
+        worksheet.write(row, 0, first_name, workbook.add_format(BORDER_TOP_BOTTOM))
+        worksheet.write(row, 1, last_name, workbook.add_format(BORDER_TOP_BOTTOM))
+
+
+def create_worksheet_points_per_exercise(workbook: Workbook, email_to_name, students_marks, all_sheet_names,
+                                         graded_sheet_names) -> defaultdict[str, list[str]]:
+    """
+    Creates an Excel worksheet that contains the points per exercise. Returns a dict with the
+    students' email addresses and the cell addresses that contain the calculated points per sheet.
+    """
+    worksheet = workbook.add_worksheet("Points Per Exercise")
+    worksheet.write(0, 0, "First Name", workbook.add_format(BOLD))
+    worksheet.write(0, 1, "Last Name", workbook.add_format(BOLD))
+    col = 2
+    for sheet_name in all_sheet_names:
+        worksheet.write(0, col, sheet_name, workbook.add_format(BOLD | BORDER_LEFT))
+        if sheet_name in graded_sheet_names:
+            col += 1
+            for exercise in sorted(graded_sheet_names[sheet_name]):
+                worksheet.write(0, col, exercise.replace("exercise", "task"), workbook.add_format(BOLD))
+                col += 1
+        else:
+            col += 1
+    all_points_per_sheet_cell_addresses = defaultdict(list)
+    student_start_row = 1
+    sorted_emails = sorted(email_to_name.keys(), key=lambda e: (email_to_name[e][0], email_to_name[e][1]))
+    for row, email in enumerate(sorted_emails, start=student_start_row):
+        add_student_name(workbook, worksheet, row, email_to_name[email])
+        points_per_sheet_cell_addresses = add_student_marks_worksheet_points_per_exercise(
+            workbook, worksheet, row, email, students_marks, all_sheet_names, graded_sheet_names
+        )
+        all_points_per_sheet_cell_addresses[email] = points_per_sheet_cell_addresses
+    worksheet.autofit()
+    return all_points_per_sheet_cell_addresses
+
+
+def create_worksheet_points_per_sheet(workbook: Workbook, _the_config: config.Config,
+                                      email_to_name, students_marks,
+                                      all_sheet_names, graded_sheet_names,
+                                      points_per_sheet_cell_addresses) -> None:
+    """
+    Creates an Excel worksheet that contains the points per sheet,
+    useful values calculated with functions, and conditional formatting.
+    """
+    worksheet = workbook.add_worksheet("Points Summary")
+    worksheet.activate()
+    worksheet.write(3 + 4, 0, "First Name", workbook.add_format(BOLD))
```

Comment:
Making the constant `4` a named variable could make this easier to read and maintain.